src/compas_fab/backends/pybullet/client.py

```diff
@@ -0,0 +1,509 @@
+            self.connection_type = 'direct'
+            print('No display detected! Continuing without GUI.')
+
+    @staticmethod
+    def _compose_options(color, width, height):
+        options = ''
+        if color is not None:
+            options += '--background_color_red={} --background_color_green={} --background_color_blue={}'.format(*color)
+        if width is not None:
+            options += '--width={}'.format(width)
+        if height is not None:
+            options += '--height={}'.format(height)
+        return options
+
+    def _configure_debug_visualizer(self, shadows):
+        pybullet.configureDebugVisualizer(pybullet.COV_ENABLE_GUI, False, physicsClientId=self.client_id)
+        pybullet.configureDebugVisualizer(pybullet.COV_ENABLE_TINY_RENDERER, False, physicsClientId=self.client_id)
+        pybullet.configureDebugVisualizer(pybullet.COV_ENABLE_RGB_BUFFER_PREVIEW, False, physicsClientId=self.client_id)
+        pybullet.configureDebugVisualizer(pybullet.COV_ENABLE_DEPTH_BUFFER_PREVIEW, False, physicsClientId=self.client_id)
+        pybullet.configureDebugVisualizer(pybullet.COV_ENABLE_SEGMENTATION_MARK_PREVIEW, False, physicsClientId=self.client_id)
+        pybullet.configureDebugVisualizer(pybullet.COV_ENABLE_SHADOWS, shadows, physicsClientId=self.client_id)
+
+    def disconnect(self):
+        """Disconnect from the PyBullet server."""
+        with redirect_stdout():
+            return pybullet.disconnect(physicsClientId=self.client_id)
+
+    @property
+    def is_connected(self):
+        """Indicates whether the client has an active connection.
+
+        Returns
+        -------
+        :obj:`bool`
+            ``True`` if connected, ``False`` otherwise.
+        """
+        if self.client_id is None:
+            return False
+        return pybullet.getConnectionInfo(physicsClientId=self.client_id)['isConnected'] == 1
+
+
+class PyBulletClient(PyBulletBase, ClientInterface):
+    """Interface to use pybullet as backend.
+
+    :class:`compasfab.backends.PyBulletClient` is a context manager type, so it's best
+    used in combination with the ``with`` statement to ensure
+    resource deallocation.
+
+    Thanks to Yijiang Huang and his work in `pybullet_planning
+    <https://github.com/yijiangh/pybullet_planning>`_ for much inspiration.
+
+    Parameters
+    ----------
+    connection_type : :obj:`str`
+        Sets the connection type. Defaults to ``'gui'``.
+    verbose : :obj:`bool`
+        Use verbose logging. Defaults to ``False``.
+
+    Examples
+    --------
+    >>> from compas_fab.backends import PyBulletClient
+    >>> with PyBulletClient(connection_type='direct') as client:
+    ...     print('Connected: %s' % client.is_connected)
+    Connected: True
+
+    """
+    def __init__(self, connection_type='gui', verbose=False):
+        super(PyBulletClient, self).__init__(connection_type)
+        self.planner = PyBulletPlanner(self)
```

Comment:
Eventually (not in this PR), this could be the point in which we hook up planners as plugins: the pybullet client could define a `@pluggable` like `create_pybullet_planner()` and by default a built-in plugin would provide it `PyBulletPlanner` instances, but if @yijiangh 's upcoming `compas_fab_pychoreo` is installed and provides another implementation, it could potentially be selected over the default one.